src/_macosx.m

```diff
@@ -1170,23 +1062,22 @@ -(void)save_figure:(id)sender
 #ifdef COMPILING_FOR_10_7
     rect = [window convertRectToBacking: rect];
 #endif
-    size = rect.size;
-    scale.width = imagesize / size.width;
-    scale.height = imagesize / size.height;
+    scale.width = imagesize / rect.size.width;
+    scale.height = imagesize / rect.size.height;
 
     rect.size.width = 32;
     rect.size.height = 32;
     rect.origin.x = gap;
     rect.origin.y = 0.5*(height - rect.size.height);
 
-    for (i = 0; i < 7; i++) {
+    for (int i = 0; i < 7; i++) {
         NSString* filename = [NSString stringWithCString: images[i]
                                                 encoding: NSUTF8StringEncoding];
         NSString* tooltip = [NSString stringWithCString: tooltips[i]
                                                encoding: NSUTF8StringEncoding];
         NSImage* image = [[NSImage alloc] initWithContentsOfFile: filename];
         buttons[i] = [[NSButton alloc] initWithFrame: rect];
-        [image setSize: size];
+        [image setSize: rect.size];
```

Comment:
... then this would be incorrect, since `rect.size` is modified between the line above and here.

Reply:
This one should be reverted now, to go along with the above one.